### DISCUSSION TITLE
모두의 정원 개발 논의

### DISCUSSION THREAD

**Discussion by yeonkkk** [woowacourse-teams/2023-pium#334] [개발 논의]
# 회의 주제
- 모두의 정원 개발을 위해 세부 사항을 논의한다.
- 참여자: 쵸파, 조이

# 핵심 기능
- 사전 식물에서 식물에 대한 버튼 표기: 게시글 수 없이 모두의 정원으로 가기 등으로 표현
- 제공하는 정보: 함께한지, 환경정보(처음 등록 시 입력하는 정보와 동일)
- 입력 받는 정보: 사용자 코멘트, 난이도 입력

# 논의 내용

## 1. api 도메인 네이밍 뭐로 할까요?
```
- comment
- garden
- management
- opinion
- board
- community
- review
- post
- tips
- garden-post
```

### 식물 이미지
- 반려 식물 vs 사전 식물 이미지? -> 반려 식물에 등록된 이미지
- 근데 사진이 꼭 필요할까요? 이미 사전 식물 이미지를 보고 들어오는데. -> 일단 데이터는 넘겨주고 사용할지 말지는 디자인하면서 결정하기

### 페이지네이션 vs 무한 스크롤
- 페이지의 장점은 페이지를 특정할 수 있다는 것이다.
- 페이지 구분이 굳이 필요하다고 느끼지 않아 무한 스크롤로 결정 (배민, 그루우 참고)
- 기본 정렬은 최신순으로 한다.

### 필터링
- 사전 식물로 필터링 할 수 있도록 한다. 
- 필터링은 검색 api를 활용한다.

## 3. 게시글 등록
- 반려식물 목록 불러오기 (내 반려식물 조회 api )

### 등록 버튼 위치
- 등록 버튼 위치에 따라 전달해야하는 정보가 다를 수 있다. 만약 반려식물 상세 조회 페이지에서 등록한다면 반려 식물 id로 바로 POST 요청이 가능하지만, 사전식물 페이지에서는 사용자의 반려식물 id를 조회한 후에 요청이 가능하다. 

## 4. 기타

- `사전식물 조회 페이지`에서는 모두의 정원으로 넘어가는 버튼만 존재 (등록 X)
- 모두의 정원에 글을 쓸 수 있는 버튼은 `모두의 정원 페이지`, `반려식물 상세 조회 페이지`에 있다. 
- 네비게이션 바에 `모두의 정원` 버튼을 추가한다. 
- 게시글 수정, 삭제는 등록과 조회가 모두 완료된 후 다시 논의한다. 

# 구현할 기능 목록

1. 사전 식물 상세 페이지: 
  모두의 정원 바로가기 클릭 -> 모두의 정원 필터링 페이지로 이동
2. 반려 식물 상세 페이지: 
  모두의 정원 게시글 등록 버튼 클릭 -> 코멘트, 난이도 입력후 등록
3. NavBar: 
  모두의 정원 페이지 버튼
4. 모두의 정원:
  모든 리뷰 보여줌
  사전 식물 검색 후 필터링
  모두의 정원에서 게시글 등록 버튼 -> 반려 식물 선택 -> 코멘트, 난이도 입력

cc. @bassyu

**Comment by yeonkkk** [comment]
- 기능 구현은 9월 15일까지 완료하는 것으로 한다.

**Comment by WaiNaat** [comment]
### 1.
`garden`에 한표합니다

### 2.
1. 이미지
게시글을 올린 사람이 본인 식물 사진을 올렸을 경우 해당 사진으로, 없으면 사전 식물 사진으로
본인의 식물을 자랑하고 싶다는 마음을 생각했을 때 사진을 보여줘야 한다고 생각해요
2. 무한스크롤
저는 언제나 페이지네이션 파지만 다수결에 따르겠습니다

### 3.
등록 버튼 위치는 '내 반려식물 상세보기' 가 맞다고 생각합니다.
자기 식물을 등록하고 싶은 거지 사전 식물을 등록하고 싶은 건 아닌 것 같아요

**Reply by WaiNaat** [reply]
추가로 지금 논의할 사항인지는 모르겠는데 회원탈퇴를 했다면 이미 작성한 글을 어떻게 할 것인지도 생각해보면 좋을듯합니다

**Comment by Choi-JJunho** [comment]
### 1. api 도메인 네이밍 뭐로 할까요?

저도 `garden`에 한표입니다!

### 2. 게시글 전체 조회

- 이미지
백엔드 API 구성 간 다음과 같은 코드처럼 흐름이 이어질 수 있을 것 같네요
```java
if(petPlant.getImageUrl() == null) {
  image = petPlant.dictionaryPlant.getImage()
}

- 페이지네이션
모바일환경에서는 스크롤을 이용한 조회가 주가 된다고 생각해서 무한스크롤 방식에 한표입니당

-
``` 
이라면 

### 3. 게시글 등록

`사전식물 페이지에서는 사용자의 반려식물 id를 조회한 후에 요청이 가능하다.`
사전식물 페이지에서 내 식물정보를 확인하기 어려울텐데 해당 화면에서 `모두의 정원에 등록하기` 와 같은 버튼이 있으면 상황에 맞지 않다고 생각이 듭니다.
참새의 의견에 동의하여 모두의 정원에 자신의 식물을 등록하고 싶은 거라고 생각합니다~


### 4. 기타

- `사전식물 조회 페이지에서는 모두의 정원으로 넘어가는 버튼만 존재 (등록 X)`
동의합니다~

- `모두의 정원에 글을 쓸 수 있는 버튼은 모두의 정원 페이지, 반려식물 상세 조회 페이지에 있다.`
동의합니다~

- `네비게이션 바에 모두의 정원 버튼을 추가한다.`
동의합니다~

- `게시글 수정, 삭제는 등록과 조회가 모두 완료된 후 다시 논의한다.`
👍

**Comment by Choi-JJunho** [comment]
모두의 정원 기능 협의 및 1차 개발 완료로 close 하겠습니다.

- https://github.com/woowacourse-teams/2023-pium/pull/365
- https://github.com/woowacourse-teams/2023-pium/pull/362
- https://github.com/woowacourse-teams/2023-pium/pull/361
- https://github.com/woowacourse-teams/2023-pium/pull/340
